sympy/concrete/zeil.py

```diff
@@ -0,0 +1,107 @@
+from sympy.simplify.radsimp import numer, denom
+from sympy.polys.polytools import Poly, lcm
+from sympy.solvers.solvers import solve
+from sympy.core.symbol import symbols
+from sympy.core.mul import prod
+from sympy.concrete.gosper import gosper_normal
+
+
+def _zeil(f, k, n, N, order):
+    k_quotient = (f.subs(k, k + 1) / f).combsimp()
+    n_quotients = [(f.subs(n, n + i) / f).combsimp() for i in range(order)]
+    numers = [numer(q) for q in n_quotients]
+    denoms = [denom(q) for q in n_quotients]
+
+    D = prod(denoms)
+    lead_quotient = (k_quotient * D / D.subs(k, k + 1)).simplify()
+    r, s, p = gosper_normal(numer(lead_quotient), denom(lead_quotient), k)
+
+    var_polys = symbols("p:{}".format(order), dummy=True)
+    poly_part = p * sum(vp * a * D / b for vp, a, b in zip(var_polys, numers, denoms))
+    poly_part = poly_part.simplify()
```

Comment:
Why is `simplify` being used here?

It is generally better to avoid the use of simplify and instead use more precise and more well defined manipulations.